File: packages/lib-model-py/src/lib_model/data/numeric_value/numeric_value_models.py
from __future__ import annotations

from enum import Enum
from typing import Optional

from pydantic import BaseModel, ConfigDict, RootModel


class NUMERICUNIT(Enum):
    basisPoint = 'basisPoint'
    billion = 'billion'
    million = 'million'
    percent = 'percent'
    thousand = 'thousand'


class NumericValueModel(BaseModel):
    model_config = ConfigDict(
        extra='allow',
    )
    unit: Optional[NUMERICUNIT] = None
    value: float


class Model(RootModel[NumericValueModel]):
    root: NumericValueModel

File: packages/lib-model-js/src/data/NumericValue/NumericValue.models.ts
export type NumericValueModel = {
  unit?: NUMERIC_UNIT;
  value: number;
};

export enum NUMERIC_UNIT {
  BASIS_POINT = 'basisPoint',
  BILLION = 'billion',
  MILLLION = 'million',
  PERCENT = 'percent',
  THOUSAND = 'thousand',
}
